odoo/addons/base/models/assetsbundle.py

```diff
@@ -102,11 +102,10 @@ def __init__(self, name, files, external_assets=(), env=None, css=True, js=True,
                     self.stylesheets.append(LessStylesheetAsset(self, **params, **css_params))
                 elif extension == 'css':
                     self.stylesheets.append(StylesheetAsset(self, **params, **css_params))
-            if js:
-                if extension == 'js':
-                    self.javascripts.append(JavascriptAsset(self, **params))
-                elif extension == 'xml':
-                    self.templates.append(XMLAsset(self, **params))
+            if js and extension == 'js':
+                self.javascripts.append(JavascriptAsset(self, **params))
+            elif extension == 'xml':
+                self.templates.append(XMLAsset(self, **params))
```

Comment:
This is not equivalent to the previous code. I can't remember why I did this. Does it still work if you undo the changes?

Reply:
I tested it and it doesn't work when changing it back, but I don't know why 🤷